modules/react-native-status/desktop/CMakeLists.txt

```diff
@@ -1,45 +1,66 @@
 
-include(${CMAKE_ROOT}/Modules/ExternalProject.cmake)
+if (WIN32)
+  #
+  # Right now we only build status-go from source for Windows, since that needs to be cross-compiled with the toolchain in Conan
+  #
+  include(${CMAKE_ROOT}/Modules/ExternalProject.cmake)
 
-if (CUSTOM_STATUSGO_BUILD_DIR_PATH)
-  set(StatusGo_ROOT ${CUSTOM_STATUSGO_BUILD_DIR_PATH})
-else()
-  set(StatusGo_ROOT "${CMAKE_CURRENT_BINARY_DIR}/StatusGo")
-endif()
-set(StatusGo_PREFIX "${StatusGo_ROOT}/src/github.com/status-im")
-set(StatusGo_SOURCE_DIR  "${StatusGo_PREFIX}/status-go")
-set(StatusGo_INCLUDE_DIR "${StatusGo_SOURCE_DIR}/build/bin")
-set(StatusGo_STATIC_LIB
-     "${StatusGo_SOURCE_DIR}/build/bin/${CMAKE_STATIC_LIBRARY_PREFIX}status${CMAKE_STATIC_LIBRARY_SUFFIX}")
-
-include_directories(${StatusGo_INCLUDE_DIR})
-
-set(CONFIGURE_SCRIPT build-status-go.sh)
-
-file(STRINGS "../../../STATUS_GO_VERSION" STATUS_GO_VERSION)
-
-ExternalProject_Add(StatusGo_ep
-  PREFIX ${StatusGo_PREFIX}
-  SOURCE_DIR ${StatusGo_SOURCE_DIR}
-  URL https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-${STATUS_GO_VERSION}.zip
-      https://github.com/status-im/status-go/archive/${STATUS_GO_VERSION}.zip
-  BUILD_BYPRODUCTS ${StatusGo_STATIC_LIB}
-  CONFIGURE_COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/${CONFIGURE_SCRIPT} ${CMAKE_SYSTEM_NAME} ${GO_ROOT_PATH} ${StatusGo_ROOT} ${StatusGo_SOURCE_DIR} ${CMAKE_C_COMPILER} ${CMAKE_CXX_COMPILER}
-  BUILD_COMMAND ""
-  INSTALL_COMMAND ""
-)
-
-set(REACT_NATIVE_DESKTOP_EXTERNAL_PROJECT_DEPS ${REACT_NATIVE_DESKTOP_EXTERNAL_PROJECT_DEPS} StatusGo_ep PARENT_SCOPE)
+  find_package(Go REQUIRED)
+
+  if (CUSTOM_STATUSGO_BUILD_DIR_PATH)
+    set(StatusGo_ROOT ${CUSTOM_STATUSGO_BUILD_DIR_PATH})
+  else()
+    set(StatusGo_ROOT "${CMAKE_CURRENT_BINARY_DIR}/StatusGo")
+  endif()
+  set(StatusGo_PREFIX "${StatusGo_ROOT}/src/github.com/status-im")
+  set(StatusGo_SOURCE_DIR  "${StatusGo_PREFIX}/status-go")
+  set(StatusGo_INCLUDE_DIR "${StatusGo_SOURCE_DIR}/build/bin")
+  set(StatusGo_STATIC_LIB
+      "${StatusGo_SOURCE_DIR}/build/bin/${CMAKE_STATIC_LIBRARY_PREFIX}status${CMAKE_STATIC_LIBRARY_SUFFIX}")
+
+  include_directories(${StatusGo_INCLUDE_DIR})
+
+  set(CONFIGURE_SCRIPT build-status-go.sh)
+
+  file(STRINGS "../../../STATUS_GO_VERSION" STATUS_GO_VERSION)
+
+  ExternalProject_Add(StatusGo_ep
+    PREFIX ${StatusGo_PREFIX}
+    SOURCE_DIR ${StatusGo_SOURCE_DIR}
+    URL https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-${STATUS_GO_VERSION}.zip
+        https://github.com/status-im/status-go/archive/${STATUS_GO_VERSION}.zip
+    BUILD_BYPRODUCTS ${StatusGo_STATIC_LIB}
+    CONFIGURE_COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/${CONFIGURE_SCRIPT} ${CMAKE_SYSTEM_NAME} ${GO_ROOT_PATH} ${StatusGo_ROOT} ${StatusGo_SOURCE_DIR} ${CMAKE_C_COMPILER} ${CMAKE_CXX_COMPILER}
+    BUILD_COMMAND ""
+    INSTALL_COMMAND ""
+  )
+
+  set(REACT_NATIVE_DESKTOP_EXTERNAL_PROJECT_DEPS ${REACT_NATIVE_DESKTOP_EXTERNAL_PROJECT_DEPS} StatusGo_ep PARENT_SCOPE)
+else (WIN32)
```

Comment:
This is a weird looking `else`, is this normal?

Reply:
It's not how I would choose to do it, but it's how I've seen CMake files being written (both here and at Plex)

Reply:
Weird as fuck, but what can you do.